fiftyone/operators/cache/decorator.py

```
@@ -142,43 +160,85 @@ def decorator(func):
                 )
 
+        if residency not in _RESIDENCY_OPTIONS:
+            raise ValueError(
+                f"Invalid residency option '{residency}'. "
+                f"Valid options are: {_RESIDENCY_OPTIONS}"
+            )
+
+        # max_size is only applicable for ephemeral and hybrid residency
+        if max_size is not None and residency not in ["ephemeral", "hybrid"]:
+            raise ValueError(
+                f"max_size is only valid for ephemeral and hybrid residency."
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)